frontend/coolblogs/src/routes/SingleBlog.jsx

```diff
@@ -1,7 +1,20 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import useGlobalReducer from '../utils/useGlobalReducer';
 
 function SingleBlog() {
-  return <div> Single Blog Post </div>;
+  const { state, getSingleBlogPost } = useGlobalReducer();
+
+  useEffect(() => {
+    getSingleBlogPost();
```

Comment:
you need to pass in the postId in reducer, 

for example you have to pass in parameter thisn way:
parameter = (postId, title, content) order matters, if you don't need a parameter, pass in null,

So now you have to actually get the post id, so how do you get the postid? You use useParams() or whatever the function is to get the current number we are on in the url,

postId = some use paramsParams function while cutting the string to get just the number from it.
  `const { state, getSingleBlogPost } = useGlobalReducer(postId);`